addons/website_sale/controllers/main.py

```diff
@@ -135,7 +135,7 @@ def toggle_switchable_view(self, view_key):
             request.session.pop('website_sale_shop_layout_mode', None)
 
 
-class WebsiteSale(http.Controller):
+class WebsiteSale(PaymentPortal):
```

Comment:
You should probably only put payment specific routes under the PaymentPortal class, not the other shop ones, unless you really need the non controller methods.